server/swap/swap.go

```diff
@@ -983,12 +983,11 @@ func (s *Swapper) processBlock(ctx context.Context, block *blockNotification) {
 			return
 		}
 
-		// Lock the matchTracker so the following checks and updates are atomic
-		// with respect to Status.
 		match.mtx.RLock()
-		defer match.mtx.RUnlock()
+		status := match.Status
+		match.mtx.RUnlock()
 
-		switch match.Status {
+		switch status {
```

Comment:
Does it make sense to use `defer` here ? my linter complains about it, and it seems that `match.Status` is all that we really need to access in this `for` loop (based on `matchTracker.mtx` comment), so I've trimmed down mutex usage in this case,

unless there is another reason to hold this mutex until this function terminates ?

Reply:
Just for the reason in the comment, to make sure that any other goroutine viewing `matchTracker.Status` and `swapStatus.swapConfirmed` (i.e. `checkInactionEventBased` and `checkInactionBlockBased`) has a consistent view of those things.

It probably doesn't matter, otherwise what's the point of a separate mutex for the `swapStatus`, but it needs to be examined carefully.

Reply:
Revert this please and we can get this in.

Reply:
Done.